### PR TITLE
CUMULUS-2434-2:Update CMR S3 Link type based on UMM-G version

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-2434: As a Cumulus integrator, I want the Cumulus to add CMR links to the S3 location](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2434)

## Changes

* Update CMR S3 Link type based on UMM-G version, UMM-G 1.6.2 or later will have 'GET DATA VIA DIRECT ACCESS' type for s3 link of science file

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,26 +8,21 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
   - Configures the example deployment to enable this feature.
 
-
 ### Changed
 
 - **CUMULUS-2434**
   - Updated `@cumulus/cmrjs` `updateCMRMetadata` and related functions to add both HTTPS URLS and S3 URIs to CMR metadata.
   - Updated `update-granules-cmr-metadata-file-links` task to add both HTTPS URLs and S3 URIs to the OnlineAccessURLs field of CMR metadata. The task configuration parameter `cmrGranuleUrlType` now has default value `both`.
-  - To use the updated `update-granules-cmr-metadata-file-links` task, the granule UMM-G metadata should have version 1.6.2 or later, since CMR s3 link type 'GET DATA VIA DIRECT ACCESS' is not valid until UMM-G version [1.6.2](https://cdn.earthdata.nasa.gov/umm/granule/v1.6.2/umm-g-json-schema.json)
+  - With the updated `update-granules-cmr-metadata-file-links` task, only the granule UMM-G metadata with version 1.6.2 or later will have the s3 link with type 'GET DATA VIA DIRECT ACCESS' for its science file. Since CMR s3 link type 'GET DATA VIA DIRECT ACCESS' is not valid until UMM-G version [1.6.2](https://cdn.earthdata.nasa.gov/umm/granule/v1.6.2/umm-g-json-schema.json)
```
Comment (dopeters) [inline CHANGELOG.md:25]: Super minor nit

```suggestion
  - With the updated `update-granules-cmr-metadata-file-links` task, only the granule UMM-G metadata with version 1.6.2 or later will have the s3 link with type 'GET DATA VIA DIRECT ACCESS' for its science file. CMR s3 link type 'GET DATA VIA DIRECT ACCESS' is not valid until UMM-G version [1.6.2](https://cdn.earthdata.nasa.gov/umm/granule/v1.6.2/umm-g-json-schema.json)
```

**Review by dopeters — CHANGES_REQUESTED** [review]
This looks really great. Just a small question about the version comparison

File: packages/cmrjs/src/cmr-utils.js [hunk 2]
```diff
@@ -596,6 +606,23 @@ async function uploadUMMGJSONCMRFile(metadataObject, cmrFile) {
...
+ * @param {Object} metadataObject - the UMMG metadata object
+ * @returns {boolean} indicate if direct s3 access type is used
+ */
+function shouldUseDirectS3Type(metadataObject) {
+  const versionWithDirectS3Type = 1.62;
+  const versionString = ummVersion(metadataObject);
+  // convert version string like 1.6.1 to 1.61 for comparision
+  if (Number(versionString.replace('.', '_').replace(/\./g, '').replace('_', '.')) >= versionWithDirectS3Type) {
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/cmrjs/src/cmr-utils.js:622]: UMM-G version doesn't follow semantic versioning.  It has versions like 1.6 which is not valid semantic version.

File: CHANGELOG.md (hunk 1 above)
Comment (jennyhliu) [reply CHANGELOG.md]: updated

File: packages/cmrjs/src/cmr-utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/cmrjs/src/cmr-utils.js:622]: The code I wrote is not great but works.  "1.6.3" -> 1.63, "1.5" -> 1.5

**Review by dopeters — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**